lib/ansible/module_utils/basic.py

```diff
@@ -524,6 +524,35 @@ def is_executable(path):
             or stat.S_IXGRP & os.stat(path)[stat.ST_MODE]
             or stat.S_IXOTH & os.stat(path)[stat.ST_MODE])
 
+def get_remote_temp():
+
+    ''' Return the remote temp path where modules are stored '''
+
+    # If using test-module and explode, the remote temp will resemble ...
+    #   /tmp/test_module_scratch/debug_dir/ansible/module_utils/basic.py
+    # If using ansible or ansible-playbook with a remote system ...
+    #   /tmp/ansible_vmweLQ/ansible_modlib.zip/ansible/module_utils/basic.py
+
+    # the path to basic.py
+    thisfile = __file__
+
+    remote_temp = None
+
+    # list.index('ansible') in reverse
+    directories = thisfile.split(os.sep)
+    directories = [x for x in reversed(directories)]
+    ansible_index = None
+    for idx,x in enumerate(directories):
+        if x == 'ansible':
+            ansible_index = idx
+            break
+    if ansible_index:
+        # Keep everything up to the index
+        directories = directories[(ansible_index + 1):]        
+        directories = [x for x in reversed(directories)]
+        remote_temp = os.sep.join(directories)
+    return remote_temp
+ 
```

Comment:
The pythonpath set in os.environ will contain either "foo/bar/tmp/ansible_modlib.zip" or "foo/bar/tmp/debug_dir". Those two directories are what I'm attempting to search for with the get_remote_temp() function.